client/sysinfo/src/sysinfo.rs

```diff
@@ -32,14 +33,117 @@ use std::{
 	time::{Duration, Instant},
 };
 
+/// Used to represent the unit used in the benchmarks.
+pub enum Unit {
+	GiBs,
+	MiBs,
+	KiBs,
+}
+
+impl fmt::Display for Unit {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		f.write_str(match self {
+			Unit::GiBs => "GiBs",
+			Unit::MiBs => "MiBs",
+			Unit::KiBs => "KiBs",
+		})
+	}
+}
+
+/// Throughput as measured in bytes per second.
+#[derive(Debug, Clone, Copy, PartialEq, PartialOrd)]
+pub struct Throughput(f64);
+
+const KIBIBYTE: f64 = 1024.0;
```

Comment:
```suggestion
const KIBIBYTE: f64 = (1<<10) as f64;
const MEBIBYTE: f64 = (1<<20) as f64;
const GIBIBYTE: f64 = (1<<30) as f64;
```
Then you have to do less float operations later.